### PR TITLE
DataMigration service and projects

### PR DESCRIPTION
**Pull request by haitsongmsft**
New schema for azure database migration services

### CODE REVIEW

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 1]
```diff
@@ -467,6 +467,8 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_SRV" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_TXT" },
                   { "$ref": "https://schema.management.azure.com/schemas/2016-03-01/Microsoft.Insights.json#/resourceDefinitions/alertrules" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-11-15-preview/Microsoft.DataMigration.json#/resourceDefinitions/services" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-11-15-preview/Microsoft.DataMigration.json#/resourceDefinitions/services_projects" },
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: >                   [](start = 163, length = 18)

please remove the spaces.

**Comment by Tiano2017** [comment]
@haitsongmsft please fix the format issues. the rest looks good.

File: schemas/2014-04-01-preview/deploymentTemplate.json (hunk 1 above)
Comment (haitsongmsft) [reply schemas/2014-04-01-preview/deploymentTemplate.json]: will do in next iteration

File: schemas/2015-01-01/deploymentTemplate.json [hunk 2]
```diff
@@ -464,6 +464,8 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_SOA" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_SRV" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_TXT" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-11-15-preview/Microsoft.DataMigration.json#/resourceDefinitions/services" }, 
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-11-15-preview/Microsoft.DataMigration.json#/resourceDefinitions/services_projects" },
```
[the remark this answers is not in the document]
Comment (haitsongmsft) [reply schemas/2015-01-01/deploymentTemplate.json]: sure. thanks

**Comment by haitsongmsft** [comment]
Changed made to remove the whitespaces.

**Pull request merged**